src/components/newProposal/actions/MintTokensInput.tsx

```diff
@@ -137,44 +146,86 @@ const MintListItem = ({
-        type="text"
-        id="address"
-        error={errors?.address ?? undefined}
-        title="An address starting with 0x, followed by 40 address characters"
-        pattern={AddressPattern}
-        className="w-full basis-2/5"
-        placeholder="0x..."
-      />
-    </ErrorWrapper>
-    <div className="flex w-full flex-row gap-2">
-      <ErrorWrapper name="Amount" error={errors?.amount ?? undefined}>
+  walletsPrefix: `actions.${number}.wallets`;
+  index: number;
+  getValues: UseFormGetValues<ProposalFormActions>;
+}) => {
+  const prefix = `${walletsPrefix}.${index}`;
+  return (
+    <>
+      <ErrorWrapper name="Address" error={errors?.address}>
         <Input
-          {...register(`${prefix}.amount`, { required: true })}
-          type="number"
-          id="tokens"
-          error={errors?.amount ?? undefined}
-          title="A number using a '.' as decimal place, e.g. '3.141'"
-          pattern={NumberPattern}
-          className="w-full basis-2/3"
-          min="0"
-          required
+          {...register(`${prefix}.address`, {
+            required: true,
+            pattern: {
+              value: AddressPattern,
+              message:
+                'Please enter an address starting with 0x, followed by 40 address characters',
+            },
+            // Custom validation function to prevent duplicate addresses
+            validate: (value: string) => {
+              let anyDuplicates = someUntill(
+                getValues(walletsPrefix),
+                (a) => a.address === value,
+                index
+              );
+              return !anyDuplicates || 'Duplicate address';
+            },
+          })}
+          type="text"
+          id="address"
+          error={errors?.address}
+          className="w-full basis-2/5"
+          placeholder="0x..."
         />
       </ErrorWrapper>
-      <HiXMark
-        className="h-5 w-5 cursor-pointer self-center"
-        onClick={onRemove}
-      />
-    </div>
-  </>
-);
+      <div className="flex w-full flex-row gap-2">
+        <ErrorWrapper name="Amount" error={errors?.amount}>
+          <Input
+            {...register(`${prefix}.amount`, {
+              required: true,
+              pattern: {
+                value: NumberPattern,
+                message: 'Please enter a number, e.g. 3.141',
+              },
+            })}
+            type="number"
+            id="tokens"
+            error={errors?.amount}
+            className="w-full basis-2/3"
+            min="0"
+            required
+          />
+        </ErrorWrapper>
+        <HiXMark
+          className="h-5 w-5 cursor-pointer self-center"
+          onClick={onRemove}
+        />
+      </div>
+    </>
+  );
+};
+
+/**
+ * Performs the same logical task as the standard some method for lists.
+ * However, it does not test the entire list, but from the start (index 0) untill (exclusive) the given index.
+ */
+function someUntill<T>(
```

Comment:
Typo 🤓

Reply:
Also put this in the `utils.ts` file instead